src/main/telemetry/smartport.c

```diff
@@ -707,11 +732,12 @@ void handleSmartPortTelemetry(void)
 
                     // thousands column (GPS fix status)
                     if (STATE(GPS_FIX))
-                        tmpi += 1000;
+                        tmpi = 2000;
                     if (STATE(GPS_FIX_HOME))
-                        tmpi += 2000;
+                        tmpi = 4000;
```

Comment:
Same as above. With your change GPS_FIX and GPS_FIX_HOME can't be reported together.

Reply:
Ditto.

Reply:
This is on purpose since GPS_FIX_HOME state implies necessarily that the prior  GPS_FIX  state has been successfully  cleared
Guys look  att it  practically ,  wen you know that the state GPS_FIX_HOME has been reached you know also know that  GPS_FIX is holding so the state GPS-FIX state is only of interest as long the GPS_FIX_HOME state has not been reached ....and of absolute of no interest after.
What is gained by those facts is that you can resort  to a more efficient state enumeration scheme and make room in the "Ones Column" slot  to  encode other  modes states.
within the limited available communication bandwidth

Reply:
The problem with this code is that only one flight mode is reported at any given time. You can't know armed state, angle mode, position hold, altitude hold, heading hold, head free and home lock all at the same time like you can currently (and are important to know). 

The way the code works now, each digit is treated as 3 bits. This allows for multiple flight modes, flight mode modifiers, and flight statuses to all be reported at the same time.

Also, there's the issue with backwards compatibility. To enhance the telemetry code, we should try to extend it without breaking every script that uses the current bit logic.

Finally, your assumption of fix home and home reset is wrong. I have a home reset switch on my transmitter. I can launch in one location, land in another location, reset my home position, continue my flight and then RTH in the new home location. I do this quite regularly as my launch position has tree cover where RTH would be bad.

Reply:
@teckel12  You are not keeping to the case in point GPS_FIX and GPS_FIX_HOME.
Beyond that the problem which you don't see is that the   3bit  coding of yours are  POSITIONAL that  is 1,2 and 4 and wasteful of the information bandwidth offered by a decimal based sequential enumeration which is  1,2,3,4,5,6,7,8,9 that is in excess of  more than 2 fold more.
The only condition being to use this vastly superior coding is that only mutually exclusive State should be handled within each digit or slot of tmp1.
Think of it

Reply:
@jaymal66 Bit logic allows for any and all states to be know, your method only allows for one state.  Sure, you can pack in more states, but you can't know if multiple states exist at the same time (which is extremely important).

Do you understand how the bit-logic works?  It's additive, so a value of 7 for example means 4+2+1.  If the value is 6, it means 4+2, 5 would be 4+1.  So, you can know exactly which combinations exist.  One doesn't cancel the others out.  I wouldn't call what you've done as vastly superior coding.  I would call it vastly simplified, and vastly limiting at the same time.  Also, there's other parts that are just wrong.

I think we've beat this to death at this point.  I know I'm done with it.

Reply:
We have obvious communication problem..which sadly  have unduly stressed our relationship right from the start.Let  mostly perceived bygones  be bygones
I am submitting a revised version witch  some adjustments and comments and include  a single important correction  something all together which I hope will help reach some common undertandings.

Reply:
As long as you've compiled/tested your changes and it extends without breaking existing telemetry scripts I'd love to see it!

Reply:
Well I've compiled and tested a revisited version which eliminates  obvious  programming glitches and  addresses most your your more serious concerns ,
Tests  doesn't include yet any of the GPS depending modes since they require to be performed outdoors .
Your requirement that the code  have to  support  full  backwards compatibility for  all  Lua  scripts implemented by all user on all TX's at large is however IMHO neither feasible nor realistic .Providing unlimited access to all Modes as enumerated in the INAV configurator is a quantum leap which requires anyway to have the scripts adapted to the new provided functionalities.
Please provide me with push writes to upload the updated version.

Reply:
It's totally feasable as well as realistic to add several more flight modes and flight mode modifiers without breaking any existing scripts, which I offered to do for you on several occasions. It would also be quite easy. 

I can't imagine your changes being merged if they include breaking existing scripts and not following basic multiple flight mode modifier logic. We all appreciate you trying to help, but at some point it's not helping anymore, but doing quite the opposite. I'm going to leave it at that.

Reply:
W've gone over that many times ,
You can turn it anyway you like  but  the fact remains that to accommodate all modes enumerated in the INAV configurator  requires  to move much more information through tmp1 that what the present encoding can possibly handle ... something hence which  inescapably   requires  to implement generally  another a new  more efficient encoding procedure and restrict the use of old procedure for the fewer specific cases where it still make sense.
In any case the required adjustments  of the Lua scripts would be truly  minimal since the decimal digit wise encoding in tmp1  of the Information is maintained and only the association of value to mode states is changed.
So again if don't agree with the above  we need indeed  to agree that we disagree